### PR TITLE
Extended techniques info

### PR DESCRIPTION
**Pull request by Night-Pryanik**
- Added extended description of techniques;
- Raised `Average melee damage` block to the techniques block and make the first line of it bold.

default

### CODE REVIEW

**Comment by Mecares** [comment]
As far as i remember do rapid strikes 66% of the damage of normal attacks and not 66% **more**, which would be 166%.

**Comment by AMurkin** [comment]
Suggestion: separate visual markup from translatable strings.
I mean concatenate "* " and _( "i18n" ).

**Comment by OzoneH3** [comment]
> As far as i remember do rapid strikes 66% of the damage of normal attacks and not 66% more, which would be 166%.

Thats right. Code snip:
```
if( weap.has_technique( rapid_strike ) ) {
    move_cost /= 2;
    avg_dmg *= 0.66;
}
```

**Comment by Night-Pryanik** [comment]
> I mean concatenate "* " and _( "i18n" ).

Could you please describe in detail what are you suggesting?

**Comment by Night-Pryanik** [comment]
> As far as i remember do rapid strikes 66% of the damage of normal attacks and not 66% more, which would be 166%.

Corrected in 927a026.

**Comment by Coolthulhu** [comment]
I don't like it, it's huge. It is a bit more informative the first time, but afterwards it is going to bloat the description.

**Comment by Night-Pryanik** [comment]
> I don't like it, it's huge. It is a bit more informative the first time, but afterwards it is going to bloat the description.

`Brutal Strike` is much shorter of course, but players (especially new players) usually have absolutely no idea what does this means and what bonuses does this technique gives.
The survivor naginata in the attached pic has 5 techniques (if reach attack could be considered as a technique). Most other weapons have no more than 2 techniques. Description of 2 techniques shouldn't bloat the menu so heavily.

If this rationale still does not convinces you, do you have any idea on improving this?

**Comment by Coolthulhu** [comment]
> If this rationale still does not convinces you, do you have any idea on improving this?

Concise, mathematical descriptions are the best. Instead of "this item will cause knockback on crits", a standardized "Brutal attack: Crit only, Knockback 1 tile" would be more readable. Instead of "this item can stun for two turns", "Wrap attack: Stun 2 turns". "Rapid attack: 66% damage, 50% move cost".
Bonus points if those were actually generated from technique definition instead of manual.

Another option would be to somehow separate the list of techniques from their long descriptions. I'm not sure how to do that, though - in a game with mouse controls, clicking/mousing over the name would bring up the long description. We can't do that here, though.
On a side note, flags would benefit from similar treatment.

Regardless of what you do, instead of hardcoding a select bunch of cases, which should be avoided where possible, you should add a field for them in the json and describe them there.

**Comment by AMurkin** [comment]
> Could you please describe in detail what are you suggesting?

You are changing translatable strings like so:
```
  _( "This item can be used to make <info>long reach attacks</info>." )
_( "* This item can be used to make <info>long reach attacks</info>." )
```
"* " part don't need to be translated and it used for only visual markup.
If you programmatically prefix string, returned by _(), with "* " that will avoid altering of already translated strings and split i18n from visual style.

In code (i think):
```
std::stringstream line;
line << "* " << _( "This item can be used to <stat>disarm</stat> weapon wielded by opponent." );
iteminfo( "DESCRIPTION", line.str() ) )
```
instead of
```
iteminfo( "DESCRIPTION", _( "* This item can be used to <stat>disarm</stat> weapon wielded by opponent." ) )
```

**Comment by Night-Pryanik** [comment]
Reworked the info based on @Coolthulhu's suggestions.

Updated screenshots in the OP.

**Comment by BevapDin** [comment]
Having the "*" inside the translated string is standard in item.cpp - it also allows the translators to put in their own markup. And if you do string concatenation, do it directly without any temporary stream: `std::string( "* " ) + _( "..." )`.

**Comment by Night-Pryanik** [comment]
Jenkins, rebuild.

File: data/json/techniques.json [hunk 1]
```diff
@@ -61,7 +65,8 @@
             "You swing through %s and everyone nearby",
             "<npcname> swings through %s and everyone nearby"
         ],
-        "aoe" : "spin"
+        "aoe" : "spin",
+        "description" : "Attack everyone nearby, crit only, min 4 melee"
```
Comment (Coolthulhu) [inline data/json/techniques.json]: Have you tried using tags in those?
In your older descriptions, you used tags to color up the description. This increased readability somewhat.

Also, "nearby" is a bit vague. "Adjacent" would be better. As far as I recall, it doesn't attack allies, so "Attack adjacent enemies" would probably be better.

Comment (Night-Pryanik) [reply data/json/techniques.json]: > Have you tried using tags in those?
In your older descriptions, you used tags to color up the description. This increased readability somewhat.

No, I haven't, but I used tags in the .cpp. Have you seen updated pic in the OP? 
The name of the technique is blue, its description is light blue.

**Comment by AncientSion** [comment]
Is it possible to add a line break after each techniques description ?

**Pull request merged**